File: Java/src/main/java/com/adaptionsoft/games/uglytrivia/Correct.java
package com.adaptionsoft.games.uglytrivia;

public class Correct extends Base implements Action {

    public Correct(CurrentPlayer currentPlayer,
            PlayerNames names,
            Places places,
            Purses purses,
            PenaltyBox penaltyBox,
            Categories categories,
            QuestionsByCategory questions,
            boolean[] isGettingOutOfPenaltyBox) {
        super(currentPlayer,
                names,
                places,
                purses,
                penaltyBox,
                categories,
                questions,
                isGettingOutOfPenaltyBox);
    }

    @Override
    public boolean execute() {
        println();
        purses.inc();

        boolean winner = didPlayerWin();
        currentPlayer.inc();

        return winner;
    }

    private void println() {
        System.out.println("Answer was corrent!!!!");
    }

    private boolean didPlayerWin() {
        return !(purses.current() == 6);
    }

}

File: Csharp/Trivia/Correct.cs
using System;

namespace UglyTrivia
{
    public class Correct : Base, Action
    {
        public Correct(CurrentPlayer currentPlayer, PlayerNames names, Places places, Purses purses, PenaltyBox penaltyBox, Categories categories, QuestionsByCategory questions, bool[] isGettingOutOfPenaltyBox)
            : base(currentPlayer, names, places, purses, penaltyBox, categories, questions, isGettingOutOfPenaltyBox)
        {
        }

        public bool Execute()
        {
            this.Println();
            this.purses.Inc();
            bool winner = this.DidPlayerWin();
            this.currentPlayer.Inc();
            return winner;
        }

        private void Println()
        {
            Console.WriteLine("Answer was corrent!!!!");
        }

        private bool DidPlayerWin()
        {
            return !(this.purses.Current() == 6);
        }
    }
}
